src/js/state/Viewer/types.js

```diff
@@ -19,7 +21,8 @@ export type ViewerState = {|
   status: SearchStatus,
   stats: ViewerStats,
   scrollX: number,
-  scrollY: number
+  scrollY: number,
+  columnHeadersView: ColumnHeadersViewState
```

Comment:
You know, I think this should actually go in the global "user prefs" area of the state. In the future, I'd like not to persist things that seem temporal like the state of the viewer. This way we'd have to write fewer migrations. However, the column headers setting seems like it would be something we'd want to save between sessions. And it seems like something that should be either at the global or tab level. I think tab level would be good for now. Can we save it it the Layout slice?

Reply:
The current behavior is that this is stored per tab and persisted in the state (and I agree that we should be cautious adding stuff there to avoid maintenance complexity). Is that what you had in mind?

I did notice that we store the `Viewer` slice of state at both the root level, representing the current tab's view, and also nested in each of the Tabs states. As far as this component is concerned, the selectors are all using the nested Viewer state under Tabs, specifically the active tab, to read data here. I was actually curious what we use the root level Viewer for?